homeassistant/components/sensor/history_values.py

```diff
@@ -0,0 +1,369 @@
+        # Get previous values of start and end
+        p_start, p_end = self._period
+
+        # Parse templates
+        await self.update_period()
+        start, end = self._period
+
+        # Convert times to UTC
+        start = dt_util.as_utc(start)
+        end = dt_util.as_utc(end)
+        p_start = dt_util.as_utc(p_start)
+        p_end = dt_util.as_utc(p_end)
+        now = datetime.datetime.now(datetime.timezone.utc)
+
+        # Compute integer timestamps
+        start_timestamp = math.floor(dt_util.as_timestamp(start))
+        end_timestamp = math.floor(dt_util.as_timestamp(end))
+        p_start_timestamp = math.floor(dt_util.as_timestamp(p_start))
+        p_end_timestamp = math.floor(dt_util.as_timestamp(p_end))
+        now_timestamp = math.floor(dt_util.as_timestamp(now))
+
+        # If period has not changed and current time after the period end...
+        if (start_timestamp == p_start_timestamp
+                and end_timestamp == p_end_timestamp
+                and end_timestamp <= now_timestamp):
+            # Don't compute anything as the value cannot have changed
+            _LOGGER.debug("Update not necessary.")
+            return
+
+        # Get history between start and end
+        history_states = await self.hass.async_add_job(
+            history.state_changes_during_period, self.hass, start, end,
+            str(self._target_entity))
+        history_list = history_states.get(self._target_entity)
+        if history_list is None:
+            self.value = STATE_UNKNOWN
+            return
+
+        if self._type in [TYPE_LOW, TYPE_PEAK, TYPE_RANGE]:
+            value = self.get_min_max_range(history_list)[self._type]
+        if self._type == TYPE_AVG:
+            value = self.get_average(history_list, end, now)
+        if value is not None:
+            self.value = value
+
+        self.async_schedule_update_ha_state()
+
+    @property
+    def device_class(self):
+        """Return the device_class of the sensor."""
+        return self._device_class
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._friendly_name
+
+    @property
+    def should_poll(self):
+        """Return False, we schedule our own updates."""
+        return False
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self.value
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit_of_measurement of the sensor."""
+        return self._unit_of_measurement
+
+    @callback
+    def get_average(self, state_list, end, now):
+        """Calculate and return the average value of @state_list."""
+        if state_list is None:
+            return
+        total_value = 0
+        precision = 0
+        total_period = datetime.timedelta(seconds=0)
+        last_idx = len(state_list)-1
+        for state in state_list:
+            if state.state == STATE_UNKNOWN:
```

Comment:
start by filtering out the STATE_UNKNOWNS:

state_list = [state for state in state_list if state.state != STATE_UNKNOWN]

Reply:
Addressed in c89c09c by removing all non-numeric states from the list before processing.